openpathsampling/tools.py

```diff
@@ -34,18 +34,36 @@ def in_ipynb():
 except ImportError:
     is_ipynb = False
 
+last_output = None
 
 def refresh_output(output_str, print_anyway=True, refresh=True,
                    output_stream=None, ipynb_display_only=False):
+
+    global last_output
+
     if output_stream is None:
         output_stream = sys.stdout
 
-    if is_ipynb or not ipynb_display_only or print_anyway:
-        if refresh:
+    if not output_str:
+        return
+
+    if refresh:
+        if is_ipynb:
             IPython.display.clear_output(wait=True)
+        else:
+            if last_output is not None:
+                lines = len(last_output.split('\n'))
+                CURSOR_UP_ONE = '\x1b[1A'
+                ERASE_LINE = '\x1b[2K'
+
+                output_stream.write((CURSOR_UP_ONE + ERASE_LINE) * (lines - 1))
```

Comment:
Yes, I don't like the global as well, but this is more a one time check, if we run in a notebook. I actually though about an output class, that works like this:

```py
o = Output(options)
o.refresh('step 1.1')
o.refresh('step 1.2')
o.next()  # or new_cell or new_step
o.refresh('step 2.2')
o.write('-- there was an error --')
```

We could also have it be more flexible like adding a progress bar, etc... But that should be a new feature and issue.